chia/full_node/mempool_manager.py

```diff
@@ -258,7 +258,22 @@ def check_removals(
         for item in conflicting_items:
             if item in conflicts:
                 continue
-            conflict_bcs = item.bundle_coin_spends[coin_id]
+            conflict_bcs = item.bundle_coin_spends.get(coin_id)
+            if conflict_bcs is None:
+                # Check if this is an item that spends an older ff singleton
+                # version with a latest version that matches our coin ID.
+                conflict_bcs = next(
+                    (
+                        bcs
+                        for bcs in item.bundle_coin_spends.values()
+                        if bcs.latest_singleton_lineage is not None and bcs.latest_singleton_lineage.coin_id == coin_id
```

Comment:
Am I understanding this correctly, that the underlying issue is that we change `latest_singleton_lineage.coin_id` without updating the corresponding dictionary key in `item.bundle_coin_spends`.

Reply:
The `spends` index gets updated, replacing the old coin ID with the new one, but the old coin ID is kept as a key in `bundle_coin_spends` due to its relationship with `coin_spends`.
As a result, if a coin ID is not found in an item's `bundle_coin_spends` despite that item being marked as conflicting on that coin ID, then that can only mean one thing: it's a fast forward singleton spend that got rebased to the latest coin ID.